crates/client/sync/src/l2.rs

```diff
@@ -247,3 +247,209 @@ pub async fn sync(
+            backend.clone(),
+            block_conv_receiver,
+            block_importer.clone(),
+            validation.clone(),
+            Some(1),
+            telemetry,
+        ));
+
+        let mock_pre_validated_block = block_importer.pre_validate(mock_block, validation.clone()).await.unwrap();
+        block_conv_sender.send(mock_pre_validated_block).await.unwrap();
+
+        drop(block_conv_sender);
+
+        match tokio::time::timeout(std::time::Duration::from_secs(120), task_handle).await {
+            Ok(Ok(_)) => (),
+            Ok(Err(e)) => panic!("Task failed: {:?}", e),
+            Err(_) => panic!("Timeout reached while waiting for task completion"),
+        }
+
+        let applied_block = backend.get_block(&DbBlockId::BlockN(0)).unwrap();
+        assert!(applied_block.is_some(), "The block was not applied correctly");
+        let applied_block = MadaraBlock::try_from(applied_block.unwrap()).unwrap();
+
+        assert_eq!(applied_block.info.header.block_number, 0, "Block number does not match");
+        assert_eq!(applied_block.info.header.block_timestamp, 0, "Block timestamp does not match");
+        assert_eq!(applied_block.info.header.parent_block_hash, Felt::ZERO, "Parent block hash does not match");
+        assert!(applied_block.inner.transactions.is_empty(), "Block should not contain any transactions");
+        assert_eq!(
+            applied_block.info.header.protocol_version,
+            StarknetVersion::default(),
+            "Protocol version does not match"
+        );
+        assert_eq!(applied_block.info.header.sequencer_address, Felt::ZERO, "Sequencer address does not match");
+        assert_eq!(applied_block.info.header.l1_gas_price.eth_l1_gas_price, 0, "L1 gas price (ETH) does not match");
+        assert_eq!(applied_block.info.header.l1_gas_price.strk_l1_gas_price, 0, "L1 gas price (STRK) does not match");
+        assert_eq!(applied_block.info.header.l1_da_mode, L1DataAvailabilityMode::Blob, "L1 DA mode does not match");
+    }
+
+    /// Test the `l2_block_conversion_task` function.
+    ///
+    /// Steps:
+    /// 1. Initialize necessary components.
+    /// 2. Create a mock block.
+    /// 3. Send the mock block to updates_sender
+    /// 4. Call the `l2_block_conversion_task` function with the mock data.
+    /// 5. Verify the results and ensure the function behaves as expected.
+    #[rstest]
+    #[tokio::test]
+    async fn test_l2_block_conversion_task(test_setup: Arc<MadaraBackend>) {
+        let backend = test_setup;
+        let (updates_sender, updates_receiver) = mpsc::channel(100);
+        let (output_sender, mut output_receiver) = mpsc::channel(100);
+        let block_import =
+            Arc::new(BlockImporter::new(backend.clone(), &MetricsRegistry::dummy(), None, true).unwrap());
+        let validation = BlockValidationContext::new(backend.chain_config().chain_id.clone());
+
+        let mock_block = UnverifiedFullBlock {
```

Comment:
maybe we can have a helper function inside the `crates/client/block_import/src/tests/block_import_utils` which would return `UnverifiedFullBlock`. This helper might be helpful for other tests as well